src/MergeContext.js

```diff
@@ -668,7 +668,9 @@ class PullRequest {
             const prMergeSha = await GH.getReference(this._mergePath());
             const prCommit = await GH.getCommit(prMergeSha);
             // whether the PR branch has not changed
-            if (this._stagedCommit.tree.sha === prCommit.tree.sha) {
```

Comment:
I was expecting this PR to _add_ a freshness condition rather than replace the old seemingly essential one. Please restore this old check or discuss why it is no longer relevant.

Reply:
Agreed - we need to compare SHAs anyway for reliability. In practice, though I guess that single 'date' check would work in all cases except when somebody (maliciously) creates an auto commit with a different tree object.